lib/connection.js

```diff
@@ -626,7 +626,15 @@ Connection.prototype.close = function (callback) {
   this.isSocketOpen = false;
   this.log('verbose', 'Closing connection to ' + this.endpoint);
   const self = this;
+
+  // If server doesn't acknowledge the half-close within connection timeout, destroy the socket.
+  const endTimeout = setTimeout(() => {
```

Comment:
`callback` should be invoked on the timeout handler, that way we always invoke it.

Reply:
`netClient.destroy` causes the `close` event to happen which triggers `callback` so it shouldn't be needed here unless I am mistaken.

Reply:
ah, I learned something :), sounds good.